model_api/cpp/models/src/segmentation_model.cpp

```diff
@@ -204,73 +212,70 @@ std::unique_ptr<ResultBase> SegmentationModel::postprocess(InferenceResult& infR
-                        if (prob > maxProb) {
-                            classId = chId;
-                            maxProb = prob;
-                        }
-                    }  // nChannels
-
-                    result->resultImage.at<uint8_t>(rowId, colId) = classId;
-                }  // width
-            }  // height
+    cv::Mat soft_prediction;
+    if (outChannels == 1 && outTensor.get_element_type() == ov::element::i32) {
+        cv::Mat predictions(outHeight, outWidth, CV_32SC1, outTensor.data<int32_t>());
+        predictions.convertTo(soft_prediction, CV_8UC1);
+    } else if (outChannels == 1 && outTensor.get_element_type() == ov::element::i64) {
+        cv::Mat predictions(outHeight, outWidth, CV_32SC1);
+        const auto data = outTensor.data<int64_t>();
+        for (size_t i = 0; i < predictions.total(); ++i) {
+            reinterpret_cast<int32_t*>(predictions.data)[i] = int32_t(data[i]);
         }
-
-        cv::resize(result->resultImage,
-                result->resultImage,
-                cv::Size(inputImgSize.inputImgWidth, inputImgSize.inputImgHeight),
-                0,
-                0,
-                cv::INTER_NEAREST);
-
-        return std::unique_ptr<ResultBase>(result);
-    }
-    float* data = outTensor.data<float>();
-    std::vector<cv::Mat> channels;
-    for (size_t c = 0; c < outTensor.get_shape()[1]; ++c) {
-        channels.emplace_back(cv::Size{outHeight, outWidth}, CV_32F, data + c * outHeight * outWidth);
+        predictions.convertTo(soft_prediction, CV_8UC1);
+    } else if (outTensor.get_element_type() == ov::element::f32) {
+        float* data = outTensor.data<float>();
+        std::vector<cv::Mat> channels;
+        for (size_t c = 0; c < outTensor.get_shape()[1]; ++c) {
+            channels.emplace_back(cv::Size{outWidth, outHeight}, CV_32FC1, data + c * outHeight * outWidth);
+        }
+        cv::merge(channels, soft_prediction);
+        cv::resize(soft_prediction, soft_prediction, {inputImgSize.inputImgWidth, inputImgSize.inputImgHeight}, 0.0, 0.0, cv::INTER_NEAREST);
     }
-    cv::Mat soft_prediction;
-    cv::merge(channels, soft_prediction);
-    cv::Mat hard_prediction = create_hard_prediction_from_soft_prediction(soft_prediction, soft_threshold, blur_strength);
-    cv::resize(hard_prediction, hard_prediction, {inputImgSize.inputImgWidth, inputImgSize.inputImgHeight}, 0.0, 0.0, cv::INTER_NEAREST);
+    cv::Mat hard_prediction = create_hard_prediction_from_soft_prediction(
+        soft_prediction, soft_threshold, blur_strength);
+
     if (return_soft_prediction) {
-        cv::resize(soft_prediction, soft_prediction, {inputImgSize.inputImgWidth, inputImgSize.inputImgHeight}, 0.0, 0.0, cv::INTER_NEAREST);
         ImageResultWithSoftPrediction* result = new ImageResultWithSoftPrediction(infResult.frameId, infResult.metaData);
         result->resultImage = hard_prediction;
```

Comment:
`hard_prediction` may or may not be resized:
else branch of `if (return_soft_prediction)` always resizes but if the condition is true, the result depends on whether `soft_prediction` was resized which depends on `outTensor.get_element_type()`

Make it consistent

Reply:
Thanks! good catch.